src/main/java/by/epam/nickgrudnitsky/util/ExpressionHandler.java

```diff
@@ -0,0 +1,162 @@
+package by.epam.nickgrudnitsky.util;
+
+import by.epam.nickgrudnitsky.exeption.CalcException;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ExpressionHandler {
+    List<String> polishNotation = new ArrayList<>();
+    LinkedList<String> stack = new LinkedList<>();
+    String text;
+
+    public ExpressionHandler(String text) {
+        this.text = text;
+    }
+
+    public String replaceExpression() {
+        List<String> expressions = searchExpressions(text);
+
+        for (String expression : expressions) {
+            String expressionResult = count(expression);
+            insertResult(expressionResult, expression);
+            polishNotation.clear();
+            stack.clear();
+        }
+        return text;
+    }
+
+    private void insertResult(String result, String expression) {
+        text = text.replace(expression, result);
+    }
+
+    private String count(String expression) {
+        translateToPolishNotation(expression);
+        String result;
+        try {
+            result = countResult();
+        } catch (CalcException e) {
+            return "ILLEGAL STATEMENT";
+        }
+        return result;
+    }
+
+    private List<String> searchExpressions(String text) {
+        List<String> expressions = new ArrayList<>();
+        Pattern pattern = Pattern.compile(Patterns.EXPRESSION);
+        Matcher matcher = pattern.matcher(text);
+
+        while (matcher.find()) {
+            expressions.add(matcher.group());
+        }
+        return expressions;
+    }
+
+    private void translateToPolishNotation(String expression) {
```

Comment:
Try to extract some methods. It is too complicated. Readability will increase if every step of counting will be named